### PR TITLE
[UT] add test case for select into outfile

### LINKED ISSUE #38020: export return error
### Steps to reproduce the behavior (Required)
```sql
mysql>  select current_timestamp(), current_timestamp() from t1 into outfile "hdfs://xxx/";
ERROR 1064 (HY000): add chunk to file writer failed, error: Unmatched number of columns expected=2 real=1
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main-bf2b96e

### PR DESCRIPTION
**Pull request by silverbullet233**
Why I'm doing:

What I'm doing:

Fixes #38020

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/runtime/file_result_writer.cpp [hunk 1]
```diff
@@ -156,9 +159,23 @@ std::string FileResultWriter::_file_format_to_name() {
...
+    auto result_chunk = std::make_shared<Chunk>();
+    for (int i = 0; i < num_columns; ++i) {
+        auto root = _output_expr_ctxs[i]->root();
+        if (!root->is_slotref()) {
+            return Status::InternalError("Not slot ref column");
+        }
+        ASSIGN_OR_RETURN(ColumnPtr column, _output_expr_ctxs[i]->evaluate(chunk));
+        auto column_ref = dynamic_cast<ColumnRef*>(root);
+        column = _output_expr_ctxs[i]->root()->type().type == TYPE_TIME
+                         ? ColumnHelper::convert_time_column_from_double_to_str(column)
```
Comment (trueeyu) [inline be/src/runtime/file_result_writer.cpp:172]: Why is the Time type treated specially here?

Comment (silverbullet233) [reply be/src/runtime/file_result_writer.cpp:172]: > Why is the Time type treated specially here?

The TIME type is represented internally as a double and needs to be converted into a string when output externally. Here I refer to the processing logic of mysql_result_writer

**Review by liuyehcf — DISMISSED** [review]

File: be/src/runtime/file_result_writer.cpp (hunk 1 above)
Comment (letian-jiang) [reply be/src/runtime/file_result_writer.cpp:172]: there is a converter for time type. see `csv::DatetimeConverter`

**Review by letian-jiang — APPROVED** [review]

**Comment by silverbullet233** [comment]
duplicated fix with #38045, only keep test case

**Review by trueeyu — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Pull request merged**